.github/workflows/cd.yml

```diff
@@ -0,0 +1,67 @@
+name: Continuous Deployment
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group:  ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  cd:
+    name: CD
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        lfs: false
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.8'
+
+    - name: Cache dependency
+      id: cache-dependencies
+      uses: actions/cache@v4
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e .[dev]
+
+    - name: Install and run pre-commit hooks
+      run: |
+        pre-commit install
+        pre-commit run --all-files --show-diff-on-failure --color=always
+
+    - name: Check types with pyright
+      run: |
+        pyright sbi
+
+    - name:  Run the fast and the slow tests with coverage
+      run: |
+        pytest -n auto -v -m "not gpu" --cov=sbi --cov-report=xml --exitfirst tests/
+
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v4-beta
+      with:
+        env_vars: OS,PYTHON
+        fail_ci_if_error: false
```

Comment:
we might switch this back to true for the deployment case, what do you think?

Reply:
you mean line 61, right? we surely could. this might be good if the CD workflow later includes also stuff like generating and publishing the docs which we might not want to do if the tests fail...

Reply:
this is sth that can go into the `codecov.yaml` right?

Reply:
yes, options like `fail_ci_if_error` would go there. I don't know about the `env_vars` though. for what is it used?